src/index.ts

```diff
@@ -199,43 +222,43 @@ const init = async (mod: Spotfire.Mod) => {
 
         // Extract data from dataview
         const fontSizeSet = fontSizeAxisMeta.parts.length > 0,
-            categoricalValue = (dvcv: DataViewCategoricalValue) => dvcv.value().map(v => v.key).reduce((res, val) => res + "," + val),
+            categoricalValue = (dvcv: DataViewCategoricalValue) => dvcv.formattedValue(","),
             tooltipGen = await createTooltipGenerator([wordsAxisMeta, fontSizeAxisMeta, colorAxisMeta]);
 
         words = rows.map((r: DataViewRow) => ({
             text: categoricalValue(r.categorical("Words")), 
             size: fontSizeSet ? r.continuous("Font size").value() as number: 20,
             color: r.color().hexCode,
             tooltip: tooltipGen(r),
+            id: r.elementId(),
             row: r
         }));
 
-        // Add id's to rows, TIBCO doesn't want me to use theirs
         let hasher = hashcode.hashCode();
-        //@ts-ignore
-        words.forEach(w => { if(!w.row.id) w.row.id = hasher.value([w.text, w.size, colorAxisMeta.parts.length == 0 ? null : colorAxisMeta.isCategorical ? categoricalValue(w.row.categorical("Color")) : w.row.continuous("Color").value()]); } );
```

Comment:
It is important not to mutate our instances. The objects should be seen as frozen.